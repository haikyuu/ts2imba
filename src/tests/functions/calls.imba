import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
Buffer.from(output, "binary")
'''

test 'simple calls work as expected' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot!
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()

let tsx-code2 = '''
res.writeHead(409, { [headers.xInertiaLocation]: req.url }).end()
'''

test 'simple calls work as expected 2' do
	const result = await build tsx-code2
	expect(result.code).toMatchSnapshot!
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()
let tsx-code3 = '''
res.status(_statusCode)
.set({
	...req.headers,
	..._headers,
	'Content-Type': 'text/html',
})
.send(html(_page, _viewData));'''

test 'simple calls work as expected 3' do
	const result = await build tsx-code3
	expect(result.code).toMatchSnapshot!
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()
