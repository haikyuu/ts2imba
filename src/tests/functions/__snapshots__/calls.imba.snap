// Vitest Snapshot v1

exports[`simple calls work as expected 3 1`] = `
"res.status(_statusCode).set({
	...req.headers
	..._headers
	\\"Content-Type\\": \\"text/html\\"
}).send html(_page, _viewData)
"
`;
